### PR TITLE
Progress on windows modules.

### PR DESCRIPTION
**Pull request by sas**

### CODE REVIEW

File: Headers/DebugServer2/Host/QueueChannel.h [hunk 1]
```diff
@@ -15,6 +15,8 @@
 #include "DebugServer2/MessageQueue.h"
 #include "DebugServer2/Types.h"
 
+#include <string>
```
Comment (nwr) [inline Headers/DebugServer2/Host/QueueChannel.h]: why?

Comment (sas) [inline Headers/DebugServer2/Host/QueueChannel.h]: Because of lines 53-54.

File: Sources/Target/Windows/Process.cpp [hunk 2]
```diff
@@ -300,6 +300,17 @@ ErrorCode Process::enumerateSharedLibraries(
...
     sl.path = Platform::WideToNarrowString(std::wstring(nameStr, nameSize));
 
+    // The following two transforms ensure that the paths we return to the
+    // debugger look like unix paths. This shouldn't be required but LLDB seems
+    // to be having trouble with paths when the host and the remote don't use
+    // the same path separator.
+    if (sl.path.length() >= 2 && sl.path[0] >= 'A' && sl.path[0] <= 'Z' &&
+        sl.path[1] == ':')
+      sl.path.erase(0, 2);
+    for (auto &c : sl.path)
```
Comment (chancila) [inline Sources/Target/Windows/Process.cpp]: std::replace_if

**Comment by chancila** [comment]
+100

**Pull request merged**